internal/doctl/mylib.go

```diff
@@ -0,0 +1,182 @@
+package mylib
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/digitalocean/godo"
+	"github.com/pkg/errors"
+)
+
+//DoctlDependencies interface for doctl dependent functions
+type DoctlDependencies interface {
+	IsAuthenticated(token string) error
+	GetCurrentDeployment(appID string) ([]byte, error)
+	RetrieveActiveDeploymentID(appID string) (string, error)
+	RetrieveActiveDeployment(deploymentID string, appID string) ([]byte, error)
+	UpdateAppPlatformAppSpec(appID string) error
+	CreateDeployments(appID string) error
+	RetrieveFromDigitalocean() ([]byte, error)
+	RetrieveAppID(appName string) (string, error)
+	IsDeployed(appID string) error
+	ReDeploy(input string, appName string) error
+}
+
+//DoctlServices is a struct for holding doctl dependent function interface
+type DoctlServices struct {
+	Dep DoctlDependencies
+}
+
+//UpdatedRepo used for parsing json object of changed repo
+type UpdatedRepo struct {
+	Name       string
+	Repository string
+	Tag        string
+}
+
+//IsAuthenticated used for user authentication
+func (d *DoctlServices) IsAuthenticated(token string) error {
```

Comment:
```suggestion
func NewDoctlClient(token string) (*DoctlServices, error) {
```
Prefixing functions with `Is` typically means it checks if something is authenticated or not. This function appears to authenticate.

I think a good idea would be to change it into the client creation function, so it will try to authenticate and if it succeeds return a "cient" that you can call the other functions on.

Reply:
updated to NewDoctlClient(token string) (*DoctlServices, error) and used the function for creation of DoctlServices ptr.